src/widgets/AutocompleteWidget.js

```diff
@@ -16,9 +16,9 @@ const allowedAttributes = [
 ];
 
 const AutocompleteWidget = withHandlers({
-  onSelect: ({ onChange, onSelect, name }) => (value) => {
+  onSelect: ({ onChange, onSelect, name }) => (value, item) => {
     if (isFunction(onSelect)) {
-      return onSelect(value);
+      return onSelect(value, item);
```

Comment:
We should remove the `return` so passing `onSelect` doesn't prevent the `onChange` from being called.